File: annotationprocessingtesting/src/main/java/com/tickaroo/tikxml/annotationprocessing/element/path/constructor/DocumentConstructor.java
package com.tickaroo.tikxml.annotationprocessing.element.path.constructor;

import com.tickaroo.tikxml.annotation.Element;
import com.tickaroo.tikxml.annotation.Path;
import com.tickaroo.tikxml.annotation.Xml;
import com.tickaroo.tikxml.annotationprocessing.element.path.Image;

/**
 * @author Hannes Dorfmann
 */
@Xml(name = "document")
public class DocumentConstructor {

  private Image image;

  DocumentConstructor(@Path("toSkip") @Element Image image){
    this.image = image;
  }

  public Image getImage() {
    return image;
  }

  @Override public boolean equals(Object o) {
    if (this == o) return true;
    if (!(o instanceof DocumentConstructor)) return false;

    DocumentConstructor that = (DocumentConstructor) o;

    return image != null ? image.equals(that.image) : that.image == null;
  }

  @Override public int hashCode() {
    return image != null ? image.hashCode() : 0;
  }
}

File: annotationprocessingtesting-kt/src/main/java/com/tickaroo/tikxml/annotationprocessing/element/path/constructor/DocumentConstructor.kt
package com.tickaroo.tikxml.annotationprocessing.element.path.constructor

import com.tickaroo.tikxml.annotation.Element
import com.tickaroo.tikxml.annotation.Path
import com.tickaroo.tikxml.annotation.Xml
import com.tickaroo.tikxml.annotationprocessing.element.path.Image

/**
 * @author Hannes Dorfmann
 */
@Xml(name = "document")
class DocumentConstructor internal constructor(@param:Path("toSkip") @param:Element val image: Image?) {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is DocumentConstructor) return false

        val that = other as DocumentConstructor?

        return if (image != null) image == that!!.image else that!!.image == null
    }

    override fun hashCode(): Int {
        return image?.hashCode() ?: 0
    }
}
